### PR TITLE
Target namespace enable

### PR DESCRIPTION
**Pull request by ianzhang366**
this pr does the followings:
1. fix some names
2. refactor manifest.go to make sure it reuse the scafflod code from the default-resources.go
3. add `targetNamespace` and `targetManagedCluster` to the hypershiftDeployment API and make sure our logic works with it.

### CODE REVIEW

File: pkg/controllers/default-resources.go [hunk 1]
```diff
@@ -203,14 +210,14 @@ func scafoldAWSNodePoolPlatform(infraOut *aws.CreateInfraOutput) *hyp.AWSNodePoo
 	}
 }
 
-func ScaffoldNodePool(namespace string, infraId string, np *hypdeployment.HypershiftNodePools) *hyp.NodePool {
+func ScaffoldNodePool(hyd *hypdeployment.HypershiftDeployment, np *hypdeployment.HypershiftNodePools) *hyp.NodePool {
```
Comment (ianzhang366) [inline pkg/controllers/default-resources.go:223]: @jnpacker does this refactor make sense to you, or would it break any logic between hypershiftDeployment and its nodepools?

My local env is broken... so have to run this with you.

File: pkg/controllers/hypershiftdeployment_controller.go [hunk 2]
```diff
@@ -73,6 +73,7 @@ const (
 //+kubebuilder:rbac:groups="",resources=secrets,verbs=create;get;list;patch;update;watch;deletecollection
 //+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch
 //+kubebuilder:rbac:groups=hypershift.openshift.io,resources=hostedclusters;nodepools,verbs=create;delete;get;list;patch;update;watch
+//+kubebuilder:rbac:groups=work.open-cluster-management.io,resources=manifestworks,verbs=create;delete;get;list;patch;update;watch
```
Comment (ianzhang366) [inline pkg/controllers/hypershiftdeployment_controller.go]: @jnpacker here's the rbac we were talking about.

File: api/v1alpha1/hypershiftdeployment_types.go [hunk 3]
```diff
@@ -80,16 +80,24 @@ type HypershiftDeploymentSpec struct {
...
 
-	//TargetNamespace allows the children resource(like, manifestwork) to be
-	//generated in other than the hypershiftDpeloyment's namespace.
+	//TargetNamespace specify the where the children resouces(hostedcluster, nodepool)
+	//to sit in other than the hypershiftDeployment's namespace
+	// +optional
 	TargetNamespace string `json:"targetNamespace"`
 
+	//TargetManagedCluster specify which managedcluster's namespace the manifestwork would be deployed.
+	//If not specified, it uses the hypershiftDeployment.
```
Comment (jnpacker) [inline api/v1alpha1/hypershiftdeployment_types.go]: This is not clear. This is the hypershiftDeployment resource.

Comment (jnpacker) [reply api/v1alpha1/hypershiftdeployment_types.go]: I think you want, if not specified, it uses the current namespace.

File: api/v1alpha1/hypershiftdeployment_types.go [hunk 4]
```diff
@@ -80,16 +80,24 @@ type HypershiftDeploymentSpec struct {
...
-	//   InfraConfigureWithManifest = "MANIFESTWORK"
+	//   ConfigureWithManifest = "MANIFESTWORK"
 	// +kubebuilder:validation:Enum=ORPHAN;INFRA-ONLY;MANIFESTWORK
 	Override InfraOverride `json:"override,omitempty"`
 
-	//TargetNamespace allows the children resource(like, manifestwork) to be
-	//generated in other than the hypershiftDpeloyment's namespace.
+	//TargetNamespace specify the where the children resouces(hostedcluster, nodepool)
+	//to sit in other than the hypershiftDeployment's namespace
+	// +optional
```
Comment (jnpacker) [inline api/v1alpha1/hypershiftdeployment_types.go:92]: I think we want to state that we use `clusters` as the default if not provided.

File: api/v1alpha1/hypershiftdeployment_types.go [hunk 5]
```diff
@@ -80,16 +80,24 @@ type HypershiftDeploymentSpec struct {
...
 	TargetNamespace string `json:"targetNamespace"`
 
+	//TargetManagedCluster specify which managedcluster's namespace the manifestwork would be deployed.
+	//If not specified, it uses the hypershiftDeployment.
+	//The managedcluster would be the management cluster of the hostedcluster and nodepool generated
```
Comment (jnpacker) [inline api/v1alpha1/hypershiftdeployment_types.go]: Replace `The managedcluster` with `The TargetManagedCluster`  This is a little clearer.

File: pkg/controllers/default-resources.go (hunk 1 above)
Comment (jnpacker) [reply pkg/controllers/default-resources.go:223]: makes sense.

File: .gitignore [hunk 6]
```diff
@@ -6,6 +6,7 @@
 *.so
 *.dylib
 bin
+pkg/pkg
```
Comment (jnpacker) [inline .gitignore:9]: What's this?

File: pkg/controllers/default-resources.go [hunk 7]
```diff
@@ -28,12 +28,20 @@ import (
 
 const ReleaseImage = "quay.io/openshift-release-dev/ocp-release:4.9.15-x86_64"
 
+func getTargetNamespace(hyd *hypdeployment.HypershiftDeployment) string {
+	if len(hyd.Spec.TargetNamespace) == 0 {
+		return hyd.GetNamespace()
```
Comment (jnpacker) [inline pkg/controllers/default-resources.go]: Maybe we log a message about the target namespace, since this is called by scaffold, it only runs once

**Review by jnpacker — CHANGES_REQUESTED** [review]
Just some comments about the comments 😉

File: .gitignore (hunk 6 above)
Comment (ianzhang366) [reply .gitignore:9]: Sometimes when running on my local env, it will generate this binary...

**Review by jnpacker — APPROVED** [review]
/lgtm

**Pull request merged**